### PR TITLE
Documentation: Add ggml_type value choices for KV cache data type in …

### PR DESCRIPTION
**Pull request by MichelleTanPY**
…README.

Updated README
- List out the rest of the available ggml_type data types for KV cache data type section.

*Make sure to read the [contributing guidelines](https://github.com/ggerganov/llama.cpp/blob/master/CONTRIBUTING.md) before submitting a PR*

### CODE REVIEW

File: examples/server/README.md [hunk 1]
```diff
@@ -62,8 +62,8 @@ The project is under active development, and we are [looking for feedback and co
 | `--yarn-beta-fast N` | YaRN: low correction dim or beta (default: 32.0)<br/>(env: LLAMA_ARG_YARN_BETA_FAST) |
 | `-dkvc, --dump-kv-cache` | verbose print of the KV cache |
 | `-nkvo, --no-kv-offload` | disable KV offload<br/>(env: LLAMA_ARG_NO_KV_OFFLOAD) |
-| `-ctk, --cache-type-k TYPE` | KV cache data type for K (default: f16)<br/>(env: LLAMA_ARG_CACHE_TYPE_K) |
-| `-ctv, --cache-type-v TYPE` | KV cache data type for V (default: f16)<br/>(env: LLAMA_ARG_CACHE_TYPE_V) |
+| `-ctk, --cache-type-k TYPE` | KV cache data type for K (default: f16, f32, bf16, q8_0, q4_0, q4_1, iq4_nl, q5_0, q5_1)<br/>(env: LLAMA_ARG_CACHE_TYPE_K) |
+| `-ctv, --cache-type-v TYPE` | KV cache data type for V (default: f16, f32, bf16, q8_0, q4_0, q4_1, iq4_nl, q5_0, q5_1)<br/>(env: LLAMA_ARG_CACHE_TYPE_V) |
```
Comment (slaren) [inline examples/server/README.md:66]: The list of valid values should be separate from the default value, which is always f16.

Comment (ngxson) [reply examples/server/README.md:66]: You should not manually edit this table. It is generated from `arg.cpp` and the next time we generate it, your changes will be discarded

**Review by ngxson — CHANGES_REQUESTED** [review]
Please edit `arg.cpp` as explained above

File: examples/server/README.md (hunk 1 above)
Comment (MichelleTanPY) [reply examples/server/README.md]: Got it. Thanks for reviewing, updated.

**Comment by ngxson** [comment]
I would prefer to convert `kv_cache_type_from_str` to an `std::map` or a static array, here instead of having to maintain it in 2 places. You can see an example of how `--chat-template` currently listing all supported templates.

**Comment by ngxson** [comment]
Hmm I think I'll do it in another PR because it's not that simple

**Comment by MichelleTanPY** [comment]
Oh ok then. In that case, is there other changes required for this PR ? :)

**Comment by ngxson** [comment]
No I think the current PR can be closed. In the other PR, list of supported KV cache type is now rendered dynamically.

**Pull request closed without merging**